packages/components/src/presentation-components/instance-filter-builder/Utils.ts

```diff
@@ -221,5 +221,26 @@ function isPropertyNumeric(typename: string) {
 }
 
 function isInvalidNumericValue(value: PrimitiveValue) {
-  return value.displayValue !== undefined && value.displayValue !== "" && isNaN(Number(value.value));
+  if (!value.displayValue || !value.value) {
+    return true;
+  }
+
+  if (typeof value.value !== "string") {
+    return isNaN(Number(value.value));
+  }
+
+  // This is handling our implementation of the IN operator without introducing breaking changes into the "@itwin/components-react".
+  // That is why we need to check for values in a serialized JSON object.
+  const { groupedRawValues } = deserializeDisplayValueGroupArray(value.displayValue, value.value);
```

Comment:
I think this should be done in `numericPropertyValidator` and only when operator is `Equals` or `NotEquals`